content/news/2022-07-29-bevy-0.8/index.md

```diff
@@ -0,0 +1,1131 @@
+}
+
+fn log_players(players: Query<&Players>) {
+    for player in &players {
+        info!("{player:?}");
+    }
+}
+```
+
+## Scripting / Modding Progress: Untyped ECS APIs
+
+<div class="release-feature-authors">authors: @jakobhellermann</div>
+
+Bevy officially only supports Rust as the "one true way to define app logic". We have [very good reasons for this](https://github.com/bevyengine/bevy/issues/114#issuecomment-672397351) and that philosophy likely won't change any time soon. But we _do_ want to provide the tools needed for the community to build 3rd party scripting / modding plugins for their languages of choice. 
+
+When we [released Bevy ECS V2](https://bevyengine.org/news/bevy-0-5/#bevy-ecs-v2), we intentionally built our internal ECS storage with these cases in mind. But we didn't expose public APIs that made it possible to interact with ECS data without normal Rust types.
+
+**Bevy 0.8** adds public "untyped" ECS apis that enable retrieving [lifetimed pointers](#ecs-lifetimed-pointers) to component and resource data using [`ComponentId`] instead of actual Rust types. 
+
+```rust
+let health_ptr: Ptr = world.entity(player).get_by_id(heath_component_id).unwrap();
+```
+
+These, when combined with our Reflection apis, provide the tools needed to start building scripting support!
+
+`@jakobhellermann` has [started building their own JavaScript / TypeScript plugin for Bevy](https://github.com/jakobhellermann/bevy_mod_js_scripting/blob/main/assets/scripts/debug.ts). Note that:
+1. This plugin is still very much a work in progress and is not ready to be used in projects.
+2. This is an unofficial community effort. Bevy will not be adding official JavaScript/TypeScript support.
+
+Here is a TypeScript snippet from their repository that queries Bevy ECS data from the script:
+
+```typescript
+const ballQuery = world.query({
+    components: [ballId, transformId, velocityId],
+});
+for (const item of ballQuery) {
+    let [ball, transform, velocity] = item.components;
+    velocity = velocity[0];
+
+    info(velocity.toString());
+}
+```
+
+[`ComponentId`]: https://docs.rs/bevy/0.8.0/bevy/ecs/component/struct.ComponentId.html
+
+## ECS "lifetimed pointers"
+
+<div class="release-feature-authors">authors: @TheRawMeatball, @BoxyUwU, @jakobhellermann</div>
+
+Bevy ECS has been refactored to use lifetimed, type-erased pointers instead of "raw pointers", which significantly improved the safety and legibility of our internals without compromising performance or flexibility.
+
+From a high level, this enables us to "retain" the lifetime of World borrows throughout our internals while still using type-erased APIs to support scenarios like [3rd party scripting languages](#scripting-modding-progress-untyped-ecs-apis).
+
+By retaining this lifetime, we can rely more on Rust's borrow checker to yell at us when we are doing something unsound. And, as it happens, this caught a number of soundness bugs!
+
+## ECS Query Internals Refactors
```

Comment:
Belongs directly before / adjacent to ReadOnly query conversion as it enabled that PR.

Reply:
I disagree. While these were important changes, from a user-facing perspective these are, I suspect, some of the least interesting changes (on average) and deserve one of the lowest spots in the ECS section.

Ordering by "conceptual dependencies" is useful (and should be preferred when possible), but it is a secondary concern relative to "putting the stuff most relevant to the average user first".

Reply:
I'm fine with that then.